src/mssw/java/com/lambdaherding/edi/mssw/ch02/ThreadLocalDateFormatter.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * 
+ */
+package com.lambdaherding.edi.mssw.ch02;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import javax.swing.text.DateFormatter;
+
+/**
+ * @author Mike
+ *
+ */
+public class ThreadLocalDateFormatter {
+
+	public void printDate(){
+		ThreadLocal<DateFormatter> thread = ThreadLocal.withInitial( 
+				() -> new DateFormatter(
+						new SimpleDateFormat("dd-MMM-YYYY")
+					)	
+				);
```

Comment:
yea i was a bit confused as to what i was supposed to do with `ThreadLocal`, i don't think i've ever used it before and didn't really understand why it would be useful